pkg/mcproducer/scripts/replace-kernel-module.sh

```diff
@@ -19,17 +21,23 @@ EOF
     cat "$kmm_config_file_filepath"
 }
 
-echo "before checking image tar file presence"
-if [ -e $kernel_module_image_filepath ]; then
-    echo "Image file $kernel_module_image_filepath found on the local file system, creating kmm config file"
+echo "before checking image presence"
+if [ -n "$(podman images -q $full_kernel_module_image 2> /dev/null)" ]; then
+    echo "Image $full_kernel_module_image found on the local file system, creating kmm config file"
     create_kmm_config
-    echo "running kernel-management worker image"
-    podman run --user=root --privileged -v /lib/modules:/lib/modules -v $kmm_config_file_filepath:/etc/kmm-worker/config.yaml -v $kernel_module_image_filepath:$kernel_module_image_filepath $worker_image kmod load --tarball /etc/kmm-worker/config.yaml
+    mkdir /tmp/opt
+    podman pod create --name kmm-pod
+    echo "creating init container"
+    podman create --pod kmm-pod --user=root --init-ctr=always -v /tmp/opt:/tmp/opt:Z $full_kernel_module_image /bin/sh -c "mkdir -p /tmp/opt/lib/modules; cp -R /opt/lib/modules/* /tmp/opt/lib/modules;"
```

Comment:
```suggestion
    podman create --pod kmm-pod --user=root --init-ctr=always -v /tmp/opt:/tmp/opt $full_kernel_module_image /bin/sh -c "mkdir -p /tmp/opt/lib/modules; cp -R /opt/lib/modules/* /tmp/opt/lib/modules;"
```

What does that `Z` do?